src/sentry/api/helpers/group_index/update.py

```diff
@@ -584,8 +591,22 @@ def update_groups(
         new_substatus = (
             SUBSTATUS_UPDATE_CHOICES[result.get("substatus")] if result.get("substatus") else None
         )
-        if new_substatus is None and new_status == GroupStatus.UNRESOLVED:
+        if new_status == GroupStatus.IGNORED and new_substatus is None:
+            if status_details.get("untilEscalating"):
+                new_substatus = GroupSubStatus.UNTIL_ESCALATING
+            elif any(status_details.get(key) is not None for key in IGNORED_CONDITION_FIELDS):
+                new_substatus = GroupSubStatus.UNTIL_CONDITION_MET
+            else:
+                new_substatus = GroupSubStatus.FOREVER
```

Comment:
I may have missed them, but can you make sure we have tests for these `status_details` cases (`untilEscalating` plus the 6 `IGNORED_CONDITION_FIELDS`).